### PR TITLE
Improve typescript bundle speed

### PR DESCRIPTION
**Pull request by fisker**
## Description

Improve string replace speed with `magic-string`. 18s -> 0.8s

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by thorn0** [comment]
What's with that failing test?

**Comment by fisker** [comment]
acorn/espree outdated

File: scripts/build/modify-typescript-module.mjs [hunk 1]
```diff
@@ -461,17 +474,17 @@ function modifyTypescriptModule(text) {
 
     // Dynamic `require()`s
     "ts.sys && ts.sys.require": "false",
-    "require(etwModulePath)": "undefined",
+    "require\\(etwModulePath\\)": "undefined", // Bug of `magic-string`?
```
Comment (fisker) [inline scripts/build/modify-typescript-module.mjs:477]: https://github.com/Rich-Harris/magic-string/issues/221

**Comment by jdalton** [comment]
Soo cool!

**Pull request merged**